keyboards/protokeeb/rev1/rev1.c

```diff
@@ -0,0 +1,28 @@
+/* Protokeeb Copyright 2024 A-Tech Officials (@atechofficials)
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.
+ * If not, see <https://www.gnu.org/licenses/gpl-3.0.html>.
+ */
+
+#include "quantum.h"
+
+// Override the matrix read function to include the encoder button
+void matrix_read_cols_on_row(matrix_row_t current_matrix[], uint8_t current_row) {
```

Comment:
You haven't tested this on hardware, correct?

Reply:
No, I have tested the firmware on the actual keyboard hardware, and the new encoder push button functionality code is working as intended. Both the rotary encoder and the encoder push button are functioning properly. When using VIA keymap, the user can successfully assign any keycodes to the encoder and its push-button using the [VIA configurator](https://usevia.app/)

Reply:
Given this implementation, it cant be working correctly.

If you replace matrix_read_cols_on_row with one that does not read any of the other pins, then none of the actual matrix would get scanned....

Reply:
Let me check if all the keys of the switch matrix are working or not because last time when I changed the encoder push button functionality code according to @drashna's suggestions:

>You could add an additional column, and add a local matrix_read_cols_on_row that reads from the specific pin for the encoder at the column/row spot. It would allow the encoder to act like it was hooked into the switch matrix, normally. However, it would no longer be an ortho 4x4 layout. But it would allow for proper via support for the encoder, for instance.

I only tested the encoder and the encoder push button functionality, not the switch matrix functionality.

Reply:
@zvecr, you were right – the switch matrix keys are not working. I need to figure out a way for both the switch matrix scanning and encoder push button to work as intended. I'm open to any suggestions, as my coding skills are not that refined.

Reply:
I managed to figure out the code to implement encoder push button functionality without affecting the switch matrix scanning. I tested the firmware on the keyboard hardware and everything is working as intended. All the keys of the switch matrix are functioning properly, and the encoder, as well as the encoder push button, are also working as intended. Please review the code and approve this pull request.

rev1.c

```C
#include "quantum.h"

extern matrix_row_t matrix[]; // Declare the matrix array

void matrix_scan_kb(void) {
    // Read the encoder button state and update the matrix
    if (gpio_read_pin(ENCODER_BUTTON_PIN) == 0) {
        matrix[ENCODER_BUTTON_ROW] |= (1 << ENCODER_BUTTON_COL);
    } else {
        matrix[ENCODER_BUTTON_ROW] &= ~(1 << ENCODER_BUTTON_COL);
    }
}
```